src/Illuminate/Routing/RoutingServiceProvider.php

```diff
@@ -140,7 +151,12 @@ protected function registerPsrRequest()
     protected function registerPsrResponse()
     {
         $this->app->bind(ResponseInterface::class, function () {
-            return new PsrResponse;
+            if (class_exists(NyholmPsrResponse::class)) {
+                return new NyholmPsrResponse;
+            }
+            if (class_exists(ZendPsrResponse::class)) {
+                return new ZendPsrResponse;
+            }
```

Comment:
Could you update this to:

```php
import Exception;
...
import Laminas\Diactoros\Response as LaminasPsrResponse;
...
```

```php
        $this->app->bind(ResponseInterface::class, function () {
            if (class_exists(NyholmPsrResponse::class)) {
                return new NyholmPsrResponse;
            }

            if (class_exists(LaminasPsrResponse::class)) {
                return new LaminasPsrResponse;
            }

            if (class_exists(ZendPsrResponse::class)) {
                return new ZendPsrResponse;
            }

            throw new Exception('Unable to resolve PSR response. Please install nyholm/psr7 or laminas/laminas-diactoros.');
        });
```

Reply:
Could you do something similar for `ServerRequestInterface` too. It's super important that we throw an exception (or just crash hard) if we cannot resolve the class. We can't just return nothing.

Reply:
I can absolutely do something similar on the response side as `Laminas\ Diactoros\Response` implements the `\Psr\Http\Message\ResponseInterface` that the others implement. On the request side, however, it's quite a bit more involved as a new factory must be created to fill the void left by the removal of `\Symfony\Bridge\PsrHttpMessage\Factory\DiactorosFactory`. 

Please correct me if I'm wrong but, to me, an implementation of such a bridge factory is out of scope for the framework and would be more appropriate for a package elsewhere.

In the meantime, I've added exceptions in the case of an unresolvable request or response.

Reply:
Yeh. We just need to make sure that the original code with the original optional dependencies still works. We can remove the extra guff if we want in Laravel 7.x, and only keep support for one of the packages, or whatever. :)

Reply:
Per the psr-http-message-bridge [changelog](https://github.com/symfony/psr-http-message-bridge/blob/master/CHANGELOG.md), they deprecated DiactorosFactory in favor of PsrHttpFactory as of 1.1.1 (2019-03-11).

An alternative is to add commentary in the `composer.json` files inline with this:
```
"symfony/psr-http-message-bridge": "Required to use PSR-7 bridging features (^1.2).",
"zendframework/zend-diactoros": "Required to use PSR-7 bridging features (^2.0) and you must add symfony/psr-http-message-bridge <=1.1 to your composer.json.",
```

The above can be a stop-gap solution while additional work proceeds with either Laminas directly or to update the Symfony psr-http-message-bridge. Thoughts?

Reply:
I guess we don't need to include Laminas tbh. I think what we have now is fine. 👍

Reply:
Ooo and Symfony isn't interested in adding support for Laminas: https://github.com/symfony/psr-http-message-bridge/pull/74